src/devices/src/virtio/balloon/device.rs

```diff
@@ -41,11 +41,25 @@ macro_rules! mem_of_active_device {
     };
 }
 
+macro_rules! mb_to_pages {
```

Comment:
It doesn't feel like this fn would belong in the `Balloon` structure as a whole, as it is not device related. I chose to use a macro instead of always multiplying with the constant to avoid duplicating code.

Reply:
You can have a module/file level function that would not be part of `Balloon` (or any other struct for that matter).

The dedup is fine, the suggestion here is only to use a fn instead of a macro.
```suggestion
fn mb_to_pages(amount_mb: u32) -> std::result::Result<u32, BalloonError> {
```

Reply:
Changed to `fn`s.